### DISCUSSION TITLE
v20.1.0-rc3

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3170] [Releases]
## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Release highlights

This is a big release, that's why we decided to move away from the 20.0.x versioning and go to 20.1.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 20.1.0. Tests are more than welcome now but be extra careful with production environment.

What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

Also:
- PHP 7.4 is now the minimum required version and 8.2 is now supported.
- the M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- a great improvement to EAV config cache has been added to v20.
- support for Google Analytics 4 was added.
- possibility to set backend locale per every admin user was added.

## How to update
Please check all the info in [this section of the README](https://github.com/OpenMage/magento-lts#since-openmage-1950--2010).

## Changelog
* Removed module Mage_Poll by @fballiano in https://github.com/OpenMage/magento-lts/pull/3098
* Upgraded ExtJS to 1.1.1 by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2473
* Bump openmage/dev-meta-package from 1.0.2 to 1.0.3 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3138
* Updated branches for codeql analysis workflow by @fballiano in https://github.com/OpenMage/magento-lts/pull/3141
* Regenerated PHPStan baseline for "main" branch by @fballiano in https://github.com/OpenMage/magento-lts/pull/3142
* Fixed some minor indentation problem for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3143
* Ended the request (and close the session) before core_app_run_after event by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/1592
* Added possibility to set backend locale per every admin user by @fballiano in https://github.com/OpenMage/magento-lts/pull/3087
* Used session_status() instead of $_SESSION in Mage_Core_Model_App by @fballiano in https://github.com/OpenMage/magento-lts/pull/3145
* Added check for coupon expiration date by @fballiano in https://github.com/OpenMage/magento-lts/pull/3144
* Improved global search performance by using only prefix matching. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/1596
* Avoided calling getBackendModelByFieldConfig with empty field by @fballiano in https://github.com/OpenMage/magento-lts/pull/3151
* Fixed typo in the exception message by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/3153
* Avoided unnecessary calls to string translation methods for catalog/sales rules labels by @luigifab in https://github.com/OpenMage/magento-lts/pull/3140
* Updated README.md to reflect new RFC 0002 - Release Schedule by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3139
* Bump phpstan/phpstan from 1.10.10 to 1.10.11 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3156
* Reduced getWebsiteId() calls by @luigifab in https://github.com/OpenMage/magento-lts/pull/3154
* Move login page title to controller by @luigifab in https://github.com/OpenMage/magento-lts/pull/3157
* Updated ACL of order objects and updated setActiveMenu for sales and catalog items by @luigifab in https://github.com/OpenMage/magento-lts/pull/3159
* [PHP 8.2] Fixed some deprecation warnings by @fballiano in https://github.com/OpenMage/magento-lts/pull/3155
* Reduced calls to char() method by @luigifab in https://github.com/OpenMage/magento-lts/pull/3161
* Fixed some whitespace issues in docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/3162
* Added order tracking for Google Analytics 4 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3092
* Fixed PHP warnings when session backend exits before returning. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3109
* Fixed price filter when search by non-numeric value by @sreichel in https://github.com/OpenMage/magento-lts/pull/3136
* Use default configurable attribute label for new products by @mmenozzi in https://github.com/OpenMage/magento-lts/pull/3168

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.1.0-rc2...v20.1.0-rc3

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.1.0-rc3'>v20.1.0-rc3</a>.</em>

**Comment by luigifab** [comment]
For all releases 20.1.0 and 19.5.0, I suggest to add something like this in description:

=> Composer update

Before `composer update`, you must run `composer config --json extra.enable-patching true` + `rm -rf app/code/core/Zend lib/Cm lib/Credis lib/mcryptcompat lib/Pelago lib/phpseclib lib/Zend`.

=> Also:

For v20: your website is faster with #3140, but I can be fastest if you replicate changes of the PR on your third party code/modules.

For v19 and v20: zf1-future, and _This may generate a problem with `Zend_Db_Select' statements that do not use 'Zend_Db_Expr' to quote expressions. If you see SQL errors after upgrading please remember to check for this specific issue in your code._ [discussion](https://github.com/OpenMage/magento-lts/discussions/3025#discussioncomment-5195579)

For v19 and v20: the composer commands to restore deleted modules?

Are there other big changes?

**Reply by fballiano** [reply]
I'll add this to the release notes of the rc3 ASAP (I don't have a real keyboard at the moment) ;-)

**Reply by luigifab** [reply]
For deleted modules, we also need the commands to remove files no? (because composer don't do it)

**Reply by fballiano** [reply]
I've linked the https://github.com/OpenMage/magento-lts#since-openmage-1950--2010 section in the release notes.

About the modules that have been removed, true, we would need a command to remove all the files too...

**Reply by colinmollenhour** [reply]
Here is a list of files that are untracked if you extract `v20.0.20` over `main`:

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        app/code/core/Mage/Adminhtml/Block/Backup.php
        app/code/core/Mage/Adminhtml/Block/Backup/
        app/code/core/Mage/Adminhtml/Block/Media/
        app/code/core/Mage/Adminhtml/Block/Poll/
        app/code/core/Mage/Adminhtml/controllers/Media/
        app/code/core/Mage/Adminhtml/controllers/Poll/
        app/code/core/Mage/Adminhtml/controllers/PollController.php
        app/code/core/Mage/Adminhtml/controllers/System/BackupController.php
        app/code/core/Mage/Backup/
        app/code/core/Mage/Poll/
        app/code/core/Zend/
        app/design/adminhtml/default/default/template/backup/
        app/design/adminhtml/default/default/template/media/editor.phtml
        app/design/adminhtml/default/default/template/poll/
        app/design/frontend/base/default/layout/poll.xml
        app/design/frontend/base/default/template/poll/
        app/etc/modules/Mage_Backup.xml
        app/etc/modules/Mage_Poll.xml
        app/locale/en_US/Mage_Backup.csv
        app/locale/en_US/Mage_Poll.csv
        dev/ddev/
        js/extjs/css/
        js/extjs/resources/css/README.txt
        js/extjs/resources/css/basic-dialog.css
        js/extjs/resources/css/box.css
        js/extjs/resources/css/button.css
        js/extjs/resources/css/combo.css
        js/extjs/resources/css/core.css
        js/extjs/resources/css/date-picker.css
        js/extjs/resources/css/dd.css
        js/extjs/resources/css/debug.css
        js/extjs/resources/css/editor.css
        js/extjs/resources/css/ext-all.css
        js/extjs/resources/css/form.css
        js/extjs/resources/css/grid.css
        js/extjs/resources/css/layout.css
        js/extjs/resources/css/menu.css
        js/extjs/resources/css/panel.css
        js/extjs/resources/css/qtips.css
        js/extjs/resources/css/reset-min.css
        js/extjs/resources/css/resizable.css
        js/extjs/resources/css/tabs.css
        js/extjs/resources/css/toolbar.css
        js/extjs/resources/css/tree.css
        js/extjs/resources/css/ytheme-aero.css
        js/extjs/resources/css/ytheme-galdaka.css
        js/extjs/resources/css/ytheme-gray.css
        js/extjs/resources/css/ytheme-vista.css
        js/extjs/resources/images/default/basic-dialog/
        js/extjs/resources/images/default/box/
        js/extjs/resources/images/default/dd/
        js/extjs/resources/images/default/editor/
        js/extjs/resources/images/default/form/
        js/extjs/resources/images/default/gradient-bg.gif
        js/extjs/resources/images/default/grid/
        js/extjs/resources/images/default/layout/
        js/extjs/resources/images/default/menu/
        js/extjs/resources/images/default/panel/
        js/extjs/resources/images/default/qtip/
        js/extjs/resources/images/default/s.gif
        js/extjs/resources/images/default/shadow-c.png
        js/extjs/resources/images/default/shadow-lr.png
        js/extjs/resources/images/default/shadow.png
        js/extjs/resources/images/default/shared/
        js/extjs/resources/images/default/sizer/
        js/extjs/resources/images/default/tabs/
        js/extjs/resources/images/default/toolbar/
        js/extjs/resources/images/default/window/
        js/extjs/resources/images/magento/basic-dialog/
        js/extjs/resources/images/magento/grid/
        js/extjs/resources/images/magento/layout/
        js/extjs/resources/images/magento/loading_bg.gif
        js/extjs/resources/images/magento/tabs/
        js/extjs/resources/images/magento/toolbar/
        js/lib/FABridge.js
        js/lib/flex.js
        js/mage/adminhtml/flexuploader.js
        lib/Cm/
        lib/Mage/Autoload/
        lib/Mage/Backup.php
        lib/Mage/Backup/
        lib/Magento/Autoload/
        lib/Pelago/
        lib/Zend/
        lib/mcryptcompat/
        lib/phpseclib/
```

**Comment by m-overlund** [comment]
is the RC3 "safe" (as can be) for production?
or still many bugs?

**Reply by luigifab** [reply]
That's depend on the weather... 😄
